cmd/root/root.go

```diff
@@ -3,31 +3,50 @@ package root
 
 import (
 	"fmt"
-
-	"github.com/spf13/cobra"
+	"github.com/open-sauced/pizza-cli/cmd/common"
 
 	"github.com/open-sauced/pizza-cli/cmd/auth"
 	"github.com/open-sauced/pizza-cli/cmd/bake"
+	"github.com/open-sauced/pizza-cli/cmd/insights"
 	repoquery "github.com/open-sauced/pizza-cli/cmd/repo-query"
 	"github.com/open-sauced/pizza-cli/pkg/api"
+	"github.com/spf13/cobra"
 )
 
+const ()
+
 // NewRootCommand bootstraps a new root cobra command for the pizza CLI
 func NewRootCommand() (*cobra.Command, error) {
 	cmd := &cobra.Command{
 		Use:   "pizza <command> <subcommand> [flags]",
 		Short: "OpenSauced CLI",
 		Long:  `A command line utility for insights, metrics, and all things OpenSauced`,
 		RunE:  run,
+		Args: func(cmd *cobra.Command, args []string) error {
+			endpointFlag := cmd.Flags().Lookup(common.FlagNameEndpoint)
+			if endpointFlag.Changed {
+				_, err := api.EndpointURLToKey(endpointFlag.Value.String())
+				if err != nil {
+					return err
+				}
+			}
+			betaFlag := cmd.Flags().Lookup(common.FlagNameBeta)
+			if betaFlag.Changed {
+				err := endpointFlag.Value.Set(api.BetaEndpointURL)
+				if err != nil {
+					return err
+				}
+			}
+			return nil
+		},
+		SilenceUsage: true,
```

Comment:
Every time a command returned an error, the `help` output was also printed, together with the error.
I think this is cleaner. The help output will still be shown in case the user entered invalid input for commands or passed `--help` flag.

Reply:
> Every time a command returned an error, the help output was also printed

We should still dump the help message when an error is returned since it's usually due to some misconfiguration of flags or values. The standard approach in Cobra CLIs is to print the help message along with the error so the user can have a quick loop to correcting it.